### ISSUE TITLE
Feature request: incoming events when they can't be processed fast enough

### ISSUE THREAD

**Issue by dandv** [lpinca/valvelet#1]
See https://github.com/xavi-/node-simple-rate-limiter/issues/6

**Comment by lpinca** [comment]
Quoting https://github.com/xavi-/node-simple-rate-limiter/issues/6:

> For some applications, lossy throughput is OK and it's desirable to drop entries that can't be processed in time, rather than have the application fall further and further behind with stale data.

This is true when throttling but if you are rate limiting you don't want calls to be ignored. Every call is important and must have its own response.

**Comment by dandv** [comment]
Agree. I wrote a [similar rate limiter](https://github.com/dandv/rate-limit/) myself because I didn't want events dropped.

That's why this should be an option. CC @markstos.

**Comment by lpinca** [comment]
I think it's out of scope, I would use a throttling utility in this case.
I don't like to have an option to change the utility behavior when there are other modules built for this.

**Comment by dandv** [comment]
What if the same app needs both throttling and queuing? For example, a stocks monitoring app would throttle quote updates, but needs to queue calls to a 3rd party API for stock details.

Would it take more than an `if()` call to distinguish between dropping and queuing? If not, I'd rather use one module than two (this one and a throttling one).

**Comment by lpinca** [comment]
> What if the same app needs both throttling and queuing?

I think it's not a common scenario but yes it's possible.
The problem I see here is that with this kind of throttling you shouldn't return a promise because the original function is not invoked. We can return `undefined` in this case but then the user should check if the returned value is a promise or not. I don't like that.

Anyway I'm open for suggestions.
